packages/in_app_purchase/in_app_purchase/test/in_app_purchase_connection/product_details_test.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:in_app_purchase/in_app_purchase.dart';
+import 'package:in_app_purchase/src/store_kit_wrappers/sk_product_wrapper.dart';
+
+void main() {
+  group('Constructor Tests', () {
+    test('fromSkProduct should correctly parse the price data', () {
+      final SKProductWrapper dummyProductWrapper = SKProductWrapper(
+        productIdentifier: 'id',
+        localizedTitle: 'title',
+        localizedDescription: 'description',
+        priceLocale:
+            SKPriceLocaleWrapper(currencySymbol: '\$', currencyCode: 'USD'),
+        subscriptionGroupIdentifier: 'com.group',
+        price: '13.37',
+      );
+
+      ProductDetails productDetails =
+          ProductDetails.fromSKProduct(dummyProductWrapper);
+      expect(productDetails.rawPrice, equals(13.37));
+    });
+
+    test('fromSkuDetails should correctly parse the price data', () {
+      final SkuDetailsWrapper dummyDetailWrapper = SkuDetailsWrapper(
+        description: 'description',
+        freeTrialPeriod: 'freeTrialPeriod',
+        introductoryPrice: 'introductoryPrice',
+        introductoryPriceMicros: 'introductoryPriceMicros',
+        introductoryPriceCycles: 1,
+        introductoryPricePeriod: 'introductoryPricePeriod',
+        price: '13.37',
+        priceAmountMicros: 13370000,
+        priceCurrencyCode: 'usd',
+        sku: 'sku',
+        subscriptionPeriod: 'subscriptionPeriod',
+        title: 'title',
+        type: SkuType.inapp,
+        originalPrice: 'originalPrice',
+        originalPriceAmountMicros: 1000,
+      );
+
+      ProductDetails productDetails =
```

Comment:
we should also test productDetails.currencyCode

Reply:
The main reason I only decided to test the `rawPrice` property is because it is the only one that has some additional logic in the constructors (String parsing for iOS, some math for Android).

I can of course also test `currencyCode`, but since that's nothing more than a value being set, just like all the other pre-existing properties, it does make me wonder if those should then also be tested (e.g. title, description, etc).

I'm curious if you have any thoughts on this?

Reply:
> it does make me wonder if those should then also be tested (e.g. title, description, etc).

Ideally, if we can, it would be nice to test all the properties being set with the constructor. It would be awesome if you could add those in this PR!

Reply:
Sure thing! The changes have been added.